### PR TITLE
Fixes #23843 - Exclude requests to local host from proxying

### PR DESCRIPTION
**Pull request by bastilian**
Requests to localhost shouldn't go over the proxy, to cause problems with loading resources for example for Katello pages.

### CODE REVIEW

File: lib/foreman/http_proxy.rb [hunk 1]
```diff
@@ -22,6 +26,12 @@ def http_proxied_rescue(&block)
 
     private
 
+    def local_request?(request_host)
+      request_host.starts_with?('127.') ||
+      request_host == 'localhost' ||
```
Comment (mmoll) [inline lib/foreman/http_proxy.rb:31]: Not sure how the IPv6 support is currently, but I think `::1` should get added here.

Comment (bastilian) [reply lib/foreman/http_proxy.rb:31]: It is possible that there is a request made to `::1`, added it as well to be an exception.

File: lib/foreman/http_proxy.rb [hunk 2]
```diff
@@ -8,10 +8,14 @@ def http_proxy_except_list
       Setting[:http_proxy_except_list]
     end
 
+    # Answers if this request should be proxied
     def proxy_http_request?(current_proxy, request_host, schema)
-      !http_proxy.nil? && current_proxy.nil? && !request_host.nil? &&
+      !http_proxy.nil? &&
+      current_proxy.nil? &
```
Comment (tbrisker) [inline lib/foreman/http_proxy.rb]: I think this should be `&&`

Comment (bastilian) [reply lib/foreman/http_proxy.rb]: Yep, it is now

**Review by tbrisker — APPROVED** [review]
Makes sense, thanks @bastilian !

**Pull request merged**